File: 1995/5/2-5.py
#!/usr/bin/env python

def getOne(n):
    return n % 10

def getTen(n):
    return (n % 100) / 10

def getHund(n):
    return n / 100

import sys

fin = file("prob5_in.txt", "r")
cnt = 1
while 1:
    act = fin.readline()
    if act == "":
        break
    line = fin.readline()
    if act[-1] == "\n":
        act = act[:-1]
    if line[-1] == "\n":
        line = line[:-1]
    if act == "encode":
        code = []
        print "Message",cnt,"(encoded): ",
        for i in xrange(len(line)):
            ch = ord(line[i])
            if ch > 99:
                code.append(getHund(ch))
            code.append(getTen(ch))
            code.append(getOne(ch))
        while code != []:
            sys.stdout.write(str(code.pop()))
        print
    elif act == "decode":
        msg = []
        print "Message",cnt,"(decoded): ",
        i = 0
        while i < len(line):
            ch1 = line[i]
            ch2 = line[i + 1]
            i += 2
            n = 10 * int(ch2) + int(ch1)
            if n < 23:
                n += 100
                i += 1
            msg.append(chr(n))
        while msg != []:
            sys.stdout.write(msg.pop())
        print
    cnt += 1

File: 1995/5/2-5.cpp
#include <iostream>
#include <string>
#include <stack>
using namespace std;

int getOne(int n)
{
    return (n % 10);
}

int getTen(int n)
{
    return ((n % 100) / 10);
}

int getHund(int n)
{
    return (n / 100);
}

int main()
{
    int cnt = 1;
    while (!cin.eof())
    {
        stack<char> msg;
        stack<int> code;
        string act;
        int n;
        char ch, ch1, ch2;
        cin >> act;
        cin.get(ch);
        if (act == "encode")
        {
            cout << "\nMessage " << cnt << " (encoded): ";
            while (cin.get(ch))
            {
                if (ch == '\n')
                    break;
                if ((int)ch > 99)
                    code.push(getHund((int)ch));
                code.push(getTen((int)ch));
                code.push(getOne((int)ch));
            }
            while (!code.empty())
            {
                cout << code.top();
                code.pop();
            }
        }
        else if (act == "decode")
        {
            cout << "\nMessage " << cnt << " (decoded): ";
            while (cin.get(ch1))
            {
                if (ch1 == '\n')
                    break;
                cin.get(ch2);
                n = 10 * (ch2 - '0') + (ch1 - '0');
                if (n < 23)
                {
                    cin.get(ch);
                    n+=100;
                }
                ch = (char)n;
                msg.push(ch);
            }
            while (!msg.empty())
            {
                cout << msg.top();
                msg.pop();
            }
        }
        cnt++;
    }
    cout << endl;
}
